src/main/java/seedu/address/model/module/Module.java

```diff
@@ -0,0 +1,101 @@
+package seedu.address.model.module;
+
+import static seedu.address.commons.util.AppUtil.areListsEqual;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import seedu.address.model.module.exam.Exam;
+import seedu.address.model.module.lesson.Lesson;
+
+/**
+ * Represents a Module in the ModBook.
+ * Guarantees: immutable.
+ */
+public class Module {
```

Comment:
Missing method for get upcoming lesson. Unless the plan is to save it for the next iteration.

Reply:
yup adding later on!